src/app/home/home.page.ts

```diff
@@ -74,4 +79,57 @@ export class HomePage {
       this.utilService.clearDatabase();
     }
   }
+
+  async getProfileDetails() {
+  await this.loader.showLoading("LOADER_MSG")
+    this.profileService.getProfileAndEntityConfigData().subscribe(async (mappedIds) => {
+      if (mappedIds) {
+        this.profilePayload = mappedIds;
+        this.getdata();
+      }
+    });
+  }
+
+  getdata() {
+    const urls = {
+      improvements: `${urlConfig.myImprovementsUrl}&page=${this.page}&limit=${this.limit}&search=&filter=`,
+      spotlight: `${urlConfig.spotlightUrl}&page=${this.page}&limit=${this.limit}&search=&filter=`,
+      recommendation: `${urlConfig.recommendationUrl}?page=${this.page}&limit=${this.limit}&search=`,
+    };
+
+    const fetchData = (url: string) =>
+      this.baseApiService.get(url).pipe(
+        catchError((err) => {
+          this.toastService.presentToast(err.error.message, 'danger');
+          return of({ status: 'error', result: { data: [], count: 0 } });
+        })
+      );
+
+    return combineLatest([
```

Comment:
what if one of the three apis fails? will all get failed or they continue to work on? hope this race condtion has been tested when using combineLatest

Reply:
it is been tested if one fails it doesn't effect the other two